### PR TITLE
feat(ui-mode): highlight console message in timeline on hover

### LINKED ISSUE #31279: [Bug]: Console tab in trace viewer
### Version

 1.44.1 

### Steps to reproduce

add a console log and open the trace viewer

### Expected behavior

when in console tab you should be able to click the console tab and it should highlight in the timeline just like network does

### Actual behavior

doesn't do anything, it shows that it is clickable but when you click nothing happens

### Additional context

_No response_

### Environment

```shell
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Memory: 502.19 MB / 64.00 GB
  Binaries:
    Node: 20.11.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.4 - /usr/local/bin/npm
    pnpm: 8.5.0 - /usr/local/bin/pnpm
  IDEs:
    VSCode: 1.90.0 - /usr/local/bin/code
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: ^1.44.1 => 1.44.1
```

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright/issues/31279

https://github.com/user-attachments/assets/cf6b0a72-1dc2-420e-bd0a-a994c71a7127

### CODE REVIEW

File: packages/trace-viewer/src/ui/timeline.tsx [hunk 1]
```diff
@@ -24,10 +24,12 @@ import type { ActionTraceEventInContext, MultiTraceModel } from './modelUtil';
 import './timeline.css';
 import type { Language } from '@isomorphic/locatorGenerators';
 import type { Entry } from '@trace/har';
+import type { ConsoleEntry } from './consoleTab';
 
 type TimelineBar = {
   action?: ActionTraceEventInContext;
   resource?: Entry;
+  consoleMessage?: ConsoleEntry
```
Comment (dgozman) [inline packages/trace-viewer/src/ui/timeline.tsx:32]:
```suggestion
  consoleMessage?: ConsoleEntry;
```

File: packages/trace-viewer/src/ui/timeline.tsx [hunk 2]
```diff
@@ -92,13 +96,34 @@ export const Timeline: React.FunctionComponent<{
...
+  }, [model, consoleEntries, boundaries, measure]);
 
   React.useMemo(() => {
-    for (const bar of bars)
-      bar.active = (!!highlightedAction && bar.action === highlightedAction) || (!!highlightedEntry && bar.resource === highlightedEntry);
-  }, [bars, highlightedAction, highlightedEntry]);
+    for (const bar of bars) {
+      if (highlightedAction)
+        bar.active = bar.action === highlightedAction;
+      if (highlightedEntry)
```
Comment (dgozman) [inline packages/trace-viewer/src/ui/timeline.tsx:119]:
```suggestion
      else if (highlightedEntry)
```

File: packages/web/src/components/listView.tsx [hunk 3]
```diff
@@ -143,6 +142,7 @@ export function ListView<T>({
         const rendered = render(item, index);
         return <div
           key={id?.(item, index) || index}
+          onDoubleClick={() => onAccepted?.(item, index)}
```
Comment (dgozman) [inline packages/web/src/components/listView.tsx:145]: Why this change?

**Review by dgozman — APPROVED** [review]

File: packages/web/src/components/listView.tsx (hunk 3 above)
Comment (mxschmitt) [reply packages/web/src/components/listView.tsx:145]: The `ListView`'s `onAccepted` handler was before only working when it was acting in 'select mode'. This means `selectedItem` was there. Now it also works without.
